src/main/java/frc/robot/subsystems/vision/Vision.java

```diff
@@ -0,0 +1,95 @@
+package frc.robot.subsystems.vision;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout.OriginPosition;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Constants.VisionConstants;
+import frc.robot.Constants.FieldConstants;;
+
+/**
+ * Runnable that gets AprilTag data from PhotonVision.
+ */
+public class Vision implements Runnable {
+
+  private final PhotonPoseEstimator photonPoseEstimator;
+  private final PhotonCamera photonCamera;
+  private final AtomicReference<EstimatedRobotPose> atomicEstimatedRobotPose = new AtomicReference<EstimatedRobotPose>();
+
+  public Vision(PhotonCamera cameraName, Transform3d robotToCamera) {
+    this.photonCamera = cameraName;
+    PhotonPoseEstimator photonPoseEstimator = null;
+
+    try {
+      var layout = AprilTagFields.k2024Crescendo.loadAprilTagLayoutField();
+      layout.setOrigin(OriginPosition.kBlueAllianceWallRightSide); // Sets the Origin to the Blue AllianceWall and will be flipped in the robot thread
+
+      // if (photonCamera != null) {
+      photonPoseEstimator = photonCamera != null
+          ? new PhotonPoseEstimator(layout, PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR, photonCamera, robotToCamera)
+          : null;
+      photonPoseEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
+      // }
+    } catch (Exception e) {
+      DriverStation.reportError("Failed to load AprilTagFieldLayout", e.getStackTrace());
+      // photonPoseEstimator = null;
```

Comment:
are you sure you don't want to set the pose estimator to null to indicate something incorrect is happening? Otherwise can you return a valid pose from an incorrectly loaded `AprilTagFieldLayout`?

Reply:
Based on the message we report to the driver station, it seems like the code that is likely to fail is `loadAprilTagLayoutField` on line 32? In this case, photonPoseEstimator would remain null I guess... Not sure if we expect the other calls in the try block to be able to fail.

Reply:
Yes